FilePersistence/src/version_control/Commit.cpp

```diff
@@ -103,4 +103,88 @@ bool Commit::getFileContent(QString fileName, const char*& content, int& content
 		return false;
 }
 
+QStringList Commit::regexSearchNodeData(QString id) const
+{
+	QStringList matches;
+	for (auto file : files())
+	{
+		int index = 0;
+		auto content = QString::fromUtf8(file->content());
+
+		if ((index = content.indexOf(id, index)) != -1)
```

Comment:
please don't assign and compare in a single if condition. write the assignment on the previous line.